### ISSUE TITLE
Ray.put on an objectid does not match Ray.get on non-objectids

### ISSUE THREAD

**Issue by Wapaul1** [ray-project/ray#352]
ray.get throws an exception if you call it on an argument that is not an objectid or list of objectids. However, ray.put does not throw an exception if you call it on an objectid.

**Comment by robertnishihara** [comment]
Right, currently if you do

```python
import ray
ray.init()

x = ray.put(1)
y = ray.put(x)

x2 = ray.get(y)
```

Then `x` and `x2` are the same, but `x` and `y` are different.

```python
assert x2 == x
assert x != y
```

However, as you suggest, we may want to throw an exception if someone calls `ray.put` on an object ID because it's probably being done on accident.

Note that we currently *disallow* returning an object ID from a remote function, for example,

```python
@ray.remote
def f():
  return 1

@ray.remote
def g():
  return f.remote()
```

Calling `g.remote()` fails with

```
Remote function __main__.g failed with:

Traceback (most recent call last):
  File "/Users/rkn/Workspace/ray/python/ray/worker.py", line 1637, in process_task
    store_outputs_in_objstore(return_object_ids, outputs, worker)
  File "/Users/rkn/Workspace/ray/python/ray/worker.py", line 1985, in store_outputs_in_objstore
    raise Exception("This remote function returned an ObjectID as its {}th return value. This is not allowed.".format(i))
Exception: This remote function returned an ObjectID as its 0th return value. This is not allowed.
```

If you really want to do this, then you can define `g` as follows.

```python
@ray.remote
def g():
  return [f.remote()]
```

**Comment by robertnishihara** [comment]
This was addressed by #353.
